tarofreighter/chain_porter.go

```diff
@@ -1051,3 +1086,14 @@ func (p *ChainPorter) stateStep(currentPkg sendPackage) (*sendPackage, error) {
 			currentPkg.SendState)
 	}
 }
+
+// copyPsbt creates a deep copy of a PSBT packet by serializing and
+// de-serializing it.
+func copyPsbt(pkg *psbt.Packet) (*psbt.Packet, error) {
```

Comment:
Worth adding it to btcd?

```
// Copy creates a deep copy of a packet so that the original does not get
// modified when the copy is manipulated.
func (pkg *Packet) Copy() (*Packet, error) {
	var buf bytes.Buffer
	if err := pkg.Serialize(&buf); err != nil {
		return nil, err
	}

	return NewFromRawBytes(&buf, false)
}
```

Reply:
I'll need to create a bunch of new helper functions for the PSBT stuff I'm working on anyway, so good idea to move that code there :+1:

Reply:
If we do this, then it'll also implement the existing `CopyErr` interface we have which is nice: https://github.com/lightninglabs/taro/blob/f6a581b6395fe779db7cbb5a4536a39bb0440dbe/chanutils/func.go#L36-L41